File: Baekjoon/이진탐색/1920.py
# https://www.acmicpc.net/problem/1920

import sys

input = sys.stdin.readline

def binary_search(i, N, start, end):
    if start > end:
        return 0
    
    mid = (end+start) // 2
    if i == N[mid]:
        return 1
    elif i < N[mid]:
            return binary_search(i, N, start, mid-1)
    else:
        return binary_search(i, N, mid+1, end)

n = int(input())
cnt = 0

N = list(map(int, input().split()))
N.sort()

m = int(input())
M = list(map(int, input().split()))

for i in M:
    print(binary_search(i, N, 0, len(N) - 1))

File: Baekjoon/이진탐색/1920.cpp
// https://www.acmicpc.net/problem/1920

#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int N, M;
    cin >> N;
    int A[N];
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
    sort(A, A + N);

    cin >> M;
    for (int i = 0; i < M; i++) {
        int num;
        cin >> num;
        cout << binary_search(A, A + N, num) << '\n';
    }

    return 0;
}